ADALiOS/ADALiOS/ADTokenCacheItem.m

```diff
@@ -169,4 +186,48 @@ - (NSString*)description
             [NSString adIsStringNilOrBlank:_refreshToken] ? @"(nil)" : @"(present)", _resource];
 }
 
+- (NSString *)clientId
+{
+    return _clientId;
+}
+
+- (void)setClientId:(NSString *)clientId
+{
+    _clientId = clientId;
+    [self calculateHash];
+}
+
+- (ADUserInformation *)userInformation
+{
+    return _userInformation;
+}
+
+- (void)setUserInformation:(ADUserInformation *)userInformation
+{
+    _userInformation = userInformation;
+    [self calculateHash];
```

Comment:
there are higher chances of introducing a bug if a new property is added and developer forgets to call calculateHash. Consider calling this function before returning the hash. So you would have getHashcode() or something like that.